compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -897,6 +892,7 @@ public void visit(BLangMatchClause matchClause) {
         matchClause.declaredVars.putAll(matchClause.matchPatterns.get(0).declaredVars);
 
         analyzeNode(matchClause.blockStmt, env);
+        this.matchClauseReturns = this.statementReturns;
```

Comment:
Can't we just rely on `statementReturns` now and remove `matchClauseReturns`? May have to move resetting `statementReturns` to the loop iterating over the clauses when visiting the `BLangMatchStatement`.

Reply:
`allClausesReturns = allClausesReturns && this.matchClauseReturns;`

This variable is used as shown above. I feel it is more readable to have `matchClauseReturns` rather than `statementReturns`

Reply:
Yeah what I meant was that we can rewrite `allClausesReturns = allClausesReturns && this.matchClauseReturns;` based on `this.statementReturns` too.

Re: readability, IMO it doesn't change much. I believe if-else blocks are a similar scenario, there too we just rely on `his.statementReturns`. I think it's consistent, easier to understand, and easier to do any changes if we follow the same pattern, instead of creating another global variable to just assign the value of `this.statementReturns`.